src/main/kotlin/org/koitharu/kotatsu/parsers/site/wpcomics/vi/TopTruyen.kt

```diff
@@ -0,0 +1,283 @@
+        SortOrder.POPULARITY,
+    )
+
+    override val filterCapabilities: MangaListFilterCapabilities
+        get() = MangaListFilterCapabilities(
+            isSearchSupported = true,
+            isTagsExclusionSupported = true,
+            isMultipleTagsSupported = false,
+        )
+
+    override suspend fun getFilterOptions(): MangaListFilterOptions {
+        return MangaListFilterOptions(
+            availableTags = availableTags(),
+            availableStates = EnumSet.of(MangaState.ONGOING, MangaState.FINISHED)
+        )
+    }
+
+	override suspend fun getListPage(page: Int, order: SortOrder, filter: MangaListFilter): List<Manga> {
+		val url = buildString {
+			append("https://")
+			append(domain)
+			append("/tim-truyen")
+			
+			if (filter.tags.isNotEmpty()) {
+				append("/")
+				append(filter.tags.first().key)
+			}
+			
+			val params = mutableListOf<String>()
+			
+			when (order) {
+				SortOrder.UPDATED -> params.add("sort=1")
+				SortOrder.POPULARITY -> params.add("sort=2")
+				else -> {} 
+			}
+			
+			filter.states.oneOrThrowIfMany()?.let { state ->
+				when (state) {
+					MangaState.ONGOING -> params.add("status=0")
+					MangaState.FINISHED -> params.add("status=1")
+					else -> params.add("status=2")
+				}
+			}
+			
+			if (filter.query?.isNotEmpty() == true) {
+				params.add("keyword=${filter.query.urlEncoded()}")
+			}
+			
+			if (page > 1) {
+				params.add("page=$page")
+			}
+			
+			if (params.isNotEmpty()) {
+				append("?")
+				append(params.joinToString("&"))
+			}
+		}
+		
+		val doc = webClient.httpGet(url).parseHtml()
+		return parseMangaList(doc)
+	}
+
+	private fun parseMangaList(doc: Document): List<Manga> {
+		return doc.select("div.item-manga").mapNotNull { div ->
+			val href = div.selectFirst("h3 a")?.attrAsRelativeUrl("href") ?: return@mapNotNull null
+			val tagElements = div.select("p.info a[href*=tim-truyen]")
+			val tags = tagElements.mapNotNullToSet { a ->
+				val key = a.attr("href").substringAfterLast("/")
+				availableTags().find { it.key == key }
+			}
+			
+			Manga(
+				id = generateUid(href),
+				url = href,
+				publicUrl = href.toAbsoluteUrl(div.host ?: domain),
+				coverUrl = div.selectFirst("div.image-item img")?.src().orEmpty(),
+				title = div.selectFirst("h3 a")?.text().orEmpty(),
+				altTitle = null,
+				rating = RATING_UNKNOWN,
+				tags = tags,
+				author = null,
+				state = null,
+				source = source,
+				isNsfw = isNsfwSource,
+			)
+		}
+	}
+
+	override suspend fun getDetails(manga: Manga): Manga {
+		val fullUrl = manga.url.toAbsoluteUrl(domain)
+		val doc = webClient.httpGet(fullUrl).parseHtml()
+		
+		val tagElements = doc.select("li.category.row p.detail-info a[href*=tim-truyen]")
+		val tags = tagElements.mapNotNullToSet { a ->
+			val key = a.attr("href").substringAfterLast("/")
+			availableTags().find { it.key == key }
+		}
+		
+		val description = doc.selectFirst("div.summary-content p.detail-summary")?.html().orEmpty()
```

Comment:
```suggestion
		val description = doc.selectFirst("div.summary-content p.detail-summary")?.html()
```

Description can be null, it is ok